### PR TITLE
Apply php 7.4 syntax + typed properties

### PR DESCRIPTION
**Pull request by samsonasik**
|    Q          |   A
|-------------- | ------
| QA            | yes

Apply php 7.4 syntax + typed properties

### CODE REVIEW

**Comment by samsonasik** [comment]
composer.lock seems require update or should I use specific branch source or/and target?

```bash
Error: Your lock file does not contain a compatible set of packages. Please run composer update.

  Problem 1
    - laminas/laminas-mvc is locked to version 3.3.0 and an update of this package was not requested.
    - laminas/laminas-mvc 3.3.0 requires laminas/laminas-view ^2.14 -> found laminas/laminas-view[3.0.x-dev] but it does not match the constraint. See https://getcomposer.org/dep-on-root for details and assistance.
  Problem 2
    - laminas/laminas-mvc-plugin-flashmessenger is locked to version 1.7.0 and an update of this package was not requested.
    - laminas/laminas-mvc-plugin-flashmessenger 1.7.0 requires laminas/laminas-view ^2.13.1 -> found laminas/laminas-view[3.0.x-dev] but it does not match the constraint. See https://getcomposer.org/dep-on-root for details and assistance.


  Problem 1
    - laminas/laminas-mvc is locked to version 3.3.0 and an update of this package was not requested.
    - laminas/laminas-mvc 3.3.0 requires laminas/laminas-view ^2.14 -> found laminas/laminas-view[3.0.x-dev] but it does not match the constraint. See https://getcomposer.org/dep-on-root for details and assistance.
  Problem 2
    - laminas/laminas-mvc-plugin-flashmessenger is locked to version 1.7.0 and an update of this package was not requested.
    - laminas/laminas-mvc-plugin-flashmessenger 1.7.0 requires laminas/laminas-view ^2.13.1 -> found laminas/laminas-view[3.0.x-dev] but it does not match the constraint. See https://getcomposer.org/dep-on-root for details and assistance.
    - ```

**Comment by samsonasik** [comment]
Re-rebase 2.19.x and its target to it too seems make it works.

**Comment by Ocramius** [comment]
@samsonasik one issue left:

```
 Error: src/Helper/HeadTitle.php:154:20: MissingClosureReturnType: Closure does not have a return type, expecting mixed (see https://psalm.dev/068)
```

Also, please run a `vendor/bin/psalm --update-baseline` after your changes :+1:

**Comment by samsonasik** [comment]
@Ocramius updated, I am not sure why DCO is failing

**Comment by samsonasik** [comment]
@Ocramius it seems DCO is failing after target to 2.20.x which include update changelog commit https://github.com/laminas/laminas-view/pull/111/commits/4ef5655ff2b5ec5ea7e78ca046fc5cfb7ed693c2

**Comment by samsonasik** [comment]
@Ocramius so, the 2.20.x branch doesn't has updated changelog, while 2.19.x has, should 2.20.x rebased against 2.19.x ?

**Comment by Ocramius** [comment]
@samsonasik frankly, `CHANGELOG.md` should be gone by now :|

**Comment by samsonasik** [comment]
@Ocramius the temporary solution probably manually cherry-pick changelog update https://github.com/laminas/laminas-view/pull/111/commits/4ef5655ff2b5ec5ea7e78ca046fc5cfb7ed693c2 to 2.20.x branch ?

File: src/Helper/HeadTitle.php [hunk 1]
```diff
@@ -151,15 +151,11 @@ public function getDefaultAttachOrder()
     private function getTitleItemCallback()
     {
         if (! $this->isTranslatorEnabled() || ! $this->hasTranslator()) {
-            return function ($item) {
-                return $item;
-            };
+            return fn($item) => $item;
```
Comment (Ocramius) [inline src/Helper/HeadTitle.php]: Most of these should be `static fn`?

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: src/Helper/HeadTitle.php (hunk 1 above)
Comment (samsonasik) [reply src/Helper/HeadTitle.php]: implemented https://github.com/laminas/laminas-view/pull/111/commits/a97613597d3ee45bf2162a4bf804e089bd70ba6e

Comment (Ocramius) [reply src/Helper/HeadTitle.php]: Oh, wow, one of them is problematic:

```

1) LaminasTest\View\Helper\FlashMessengerTest::testCanDisplayListOfMessages
Cannot bind an instance to a static closure
```

Comment (samsonasik) [reply src/Helper/HeadTitle.php]: should be fixed now https://github.com/laminas/laminas-view/pull/111/commits/7a5e318201da6f360f722c551da0363b80b76447

**Review by Ocramius — APPROVED** [review]
Thanks @samsonasik!

**Pull request merged**